src/dataclay/runtime/runtime.py

```diff
@@ -596,201 +691,195 @@ def new_object_replica(
-
-    def federate_all_objects(self, dest_dataclay_id):
-        raise NotImplementedError()
-
-    def register_external_dataclay(self, id, host, port):
-        """Register external dataClay for federation
-        Args:
-            host: external dataClay host name
-            port: external dataClay port
-        """
-        self.metadata_service.autoregister_mds(id, host, port)
-
-    ###########
-    # Tracing #
-    ###########
-
-    def activate_tracing(self, initialize):
-        """Activate tracing"""
-        initialize_extrae(initialize)
-
-    def deactivate_tracing(self, finalize_extrae):
-        """Close the runtime paraver manager and deactivate the traces in LM (That deactivate also the DS)"""
-        finish_tracing(finalize_extrae)
-
-    def activate_tracing_in_dataclay_services(self):
-        """Activate the traces in LM (That activate also the DS)"""
-        if extrae_tracing_is_enabled():
-            self.backend_clients["@LM"].activate_tracing(get_current_available_task_id())
-
-    def deactivate_tracing_in_dataclay_services(self):
-        """Deactivate the traces in LM and DSs"""
-        if extrae_tracing_is_enabled():
-            self.backend_clients["@LM"].deactivate_tracing()
-
-    def get_traces_in_dataclay_services(self):
-        """Get temporary traces from LM and DSs and store it in current workspace"""
-        traces = self.backend_clients["@LM"].get_traces()
-        traces_dir = settings.TRACES_DEST_PATH
-        if len(traces) > 0:
-            set_path = settings.TRACES_DEST_PATH + "/set-0"
-            trace_mpits = settings.TRACES_DEST_PATH + "/TRACE.mpits"
-            with open(trace_mpits, "a+") as trace_file:
-                # store them here
-                for key, value in traces.items():
-                    dest_path = set_path + "/" + key
-                    logger.debug("Storing object %s" % dest_path)
-                    with open(dest_path, "wb") as dest_file:
-                        dest_file.write(value)
-                        dest_file.close()
-                    if key.endswith("mpit"):
-                        pointer = str(dest_path) + " named\n"
-                        trace_file.write(pointer)
-                        logger.debug("Appending to %s line: %s" % (trace_mpits, pointer))
-                trace_file.flush()
-                trace_file.close()
+        await self.send_objects([instance], backend_id, True, recursive, remotes)
 
     ############
     # Shutdown #
     ############
 
     @abstractmethod
-    def stop(self):
+    async def stop(self):
         pass
 
-    def close_backend_clients(self):
-        """Stop connections and daemon threads."""
-        logger.debug("** Stopping runtime **")
-        for name, client in self.backend_clients.items():
-            logger.debug("Closing client connection to %s", name)
-            client.close()
-        self.backend_clients.stop_update()
+    ######################
+
+    # DEPRECATED
```

Comment:
This file has too many commented and deprecated lines. Explain in words, but non-functional non-descriptive lines should go away. (Reader/developer can explore git history).